src/serde/reader.rs

```diff
@@ -46,10 +47,8 @@ pub enum DeserializeError {
     /// If decoding a message takes more than the provided size limit, this
     /// error is returned.
     SizeLimit,
-    SyntaxError(String),
-    EndOfStreamError,
-    UnknownFieldError,
-    MissingFieldError,
+    EndOfStream,
```

Comment:
`serde::de::value::Error` has a `EndOfStreamError`, I don't think there is any need to duplicate it.